src/main/java/ee/tuleva/onboarding/payment/PaymentProviderCallbackService.java

```diff
@@ -0,0 +1,90 @@
+package ee.tuleva.onboarding.payment;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.crypto.MACVerifier;
+import ee.tuleva.onboarding.user.User;
+import ee.tuleva.onboarding.user.UserService;
+import java.math.BigDecimal;
+import java.text.ParseException;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class PaymentProviderCallbackService {
+
+  private final Map<String, PaymentProviderBankConfiguration> paymentProviderBankConfigurations;
+  private final UserService userService;
+
+  private final PaymentRepository paymentRepository;
+
+  private final ObjectMapper objectMapper;
+
+  public void processToken(String serializedToken) {
+
+    JWSObject token = parseToken(serializedToken);
+    verifyToken(token);
+
+    String serializedInternalReference =
+        token.getPayload().toJSONObject().get("merchant_reference").toString();
+
+    BigDecimal amount = new BigDecimal(token.getPayload().toJSONObject().get("amount").toString());
+
```

Comment:
Any way to map the token into a DTO first? I.e. to avoid doing 
`token.getPayload().toJSONObject().get("merchant_reference")` and `token.getPayload().toJSONObject().get("amount")`. 
So instead we could do `dto.getAmount()` and `dto.getInternalReference()`